Source/BugsnagClient.m

```diff
@@ -981,36 +1001,55 @@ - (void)notifyInternal:(BugsnagEvent *_Nonnull)event
     // handled errors should persist any information edited by the user
     // in a section within the KSCrash report so it can be read
     // when the error is delivered
-    [self.crashSentry reportUserException:eventErrorClass
-                                   reason:eventMessage
-                        originalException:exc
+    [self.crashSentry reportUserException:@""
+                                   reason:@""
                              handledState:[event.handledState toJson]
                                  appState:[self.state toDictionary]
                         callbackOverrides:event.overrides
                            eventOverrides:[event toJson]
                                  metadata:[event.metadata toDictionary]
-                                   config:[self.configuration.config toDictionary]
-                             discardDepth:depth];
+                                   config:[self.configuration.config toDictionary]];
 
     // A basic set of event metadata
     NSMutableDictionary *metadata = [@{
-            BSGKeyErrorClass : eventErrorClass,
+            BSGKeyErrorClass : event.errors[0].errorClass,
             BSGKeyUnhandled : [[event handledState] unhandled] ? @YES : @NO,
             BSGKeySeverity : BSGFormatSeverity(event.severity)
     } mutableCopy];
 
     // Only include the eventMessage if it contains something
+    NSString *eventMessage = event.errors[0].errorMessage;
     if (eventMessage && [eventMessage length] > 0) {
         [metadata setValue:eventMessage forKey:BSGKeyName];
     }
 
     [self addAutoBreadcrumbOfType:BSGBreadcrumbTypeError
-                      withMessage:eventErrorClass
+                      withMessage:event.errors[0].errorClass
                       andMetadata:metadata];
 
     [self flushPendingReports];
 }
 
+- (BugsnagError *)generateError:(NSException *)exception
+                                 threads:(NSArray<BugsnagThread *> *)threads {
+    NSString *errorClass = exception.name ?: NSStringFromClass([exception class]);
+    NSString *errorMessage = exception.reason;
+
+    BugsnagThread *errorReportingThread;
+
+    for (BugsnagThread *thread in threads) {
+        if (thread.errorReportingThread) {
+            errorReportingThread = thread;
+            break;
+        }
+    }
+
+    BugsnagError *error = [[BugsnagError alloc] initWithErrorReportingThread:errorReportingThread];
+    error.errorClass = errorClass;
+    error.errorMessage = errorMessage ?: @"";
+    return error;
+}
```

Comment:
I don't see any reason we can't use `initWithErrorClass` for this - we basically just want to set class, message and type. Passing in the `errorReportingThread` value doesn't seem to do anything with an empty `event` parameter.

Reply:
Updated to use existing constructor.